src/MICore/CommandFactories/lldb.cs

```diff
@@ -160,5 +160,12 @@ public override Task Catch(string name, bool onlyOnce = false, ResultClass resul
         {
             throw new NotImplementedException("lldb catch command");
         }
+
+        public async override Task<string> VarAssign(string variableName, string expression, int threadId, uint frameLevel)
+        {
+            Results results = await VarCreate(expression, threadId, frameLevel, 0, ResultClass.done);
+            string value = results.FindString("value");
```

Comment:
That's a good point.

For primitive types in C++, this doesn't work for `bool` or `char` because lldb doesn't treat `true`, `false`, `'a'` as valid values. The hope here was that those values would evaluate to the corresponding integer value (e.g., `'a'` --> `97`). I'll be creating a bug in the lldb database to see how we can address this.

For `char*`, it's being displayed as a tree structure in VSCode, so it won't allow to set the value.

Is there a way you recommend we can get the actual integer value for `char` or `bool` directly from lldb?

In order to avoid regressions, do you think `lldb::VarAssign` should first try the vanilla `VarAssign` (not trying to evaluate the expression), and if that fails, then try to evaluate the expression and set the result with another `VarAssign`?